Failing OAuth functional tests on master

* oauth sign up verification_redirect - signup, same browser, original tab closed, verification_redirect=always
* oauth sign up verification_redirect - signup, same browser different window, verification_redirect=always
* oauth sign up verification_redirect - signup, verify different browser, verification_redirect=always
* oauth permissions for untrusted reliers - signin with additional requested permission
* oauth permissions for trusted reliers - signup with `prompt=consent`
* oauth permissions for trusted reliers - signin with `prompt=consent`

My output: 

```
× firefox on any platform - oauth sign up verification_redirect - signup, same browser, original tab closed, verification_redirect=always (30.006s)
CancelError: Timeout reached on firefox on any platform - oauth sign up verification_redirect - signup, same browser, original tab closed, verification_redirect=always
  at null._onTimeout  <node_modules/intern/lib/Test.js:141:20>
  at Timer.listOnTimeout  <timers.js:92:15>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Command.finally  <node_modules/intern/node_modules/leadfoot/Command.js:566:15>
  at <node_modules/intern/lib/Test.js:216:37>
  at Promise.cancel  <node_modules/intern/browser_modules/dojo/Promise.ts:343:37>
  at null._onTimeout  <node_modules/intern/lib/Test.js:143:21>
  at Timer.listOnTimeout  <timers.js:92:15>
× firefox on any platform - oauth sign up verification_redirect - signup, same browser different window, verification_redirect=always (1.219s)
AssertionError: assert.fail()
  at Command.<anonymous>  <tests/functional/oauth_sign_up_verification_redirect.js:115:20>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at process._tickCallback  <node.js:365:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Test.registerSuite.signup, same browser different window, verification_redirect=always [as test]  <tests/functional/oauth_sign_up_verification_redirect.js:112:10>
  at <node_modules/intern/lib/Test.js:191:24>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/browser_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/browser_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/browser_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at process._tickCallback  <node.js:365:13>
× firefox on any platform - oauth sign up verification_redirect - signup, verify different browser, verification_redirect=always (11.437s)
ScriptTimeout: [POST http://localhost:4444/wd/hub/session/e1ef1ce4-be72-4892-9ca5-79df08429591/execute_async / {"script":"return (function (poller, args, timeout, pollInterval, done) {\n\t\t\t\t\t\t/* jshint evil:true */\n\t\t\t\t\t\tpoller = new Function(poller);\n\n\t\t\t\t\t\tvar endTime = Number(new Date()) + timeout;\n\n\t\t\t\t\t\t(function poll() {\n\t\t\t\t\t\t\tvar result = poller.apply(this, args);\n\n\t\t\t\t\t\t\t/*jshint evil:true */\n\t\t\t\t\t\t\tif (result != null) {\n\t\t\t\t\t\t\t\tdone(result);\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\telse if (Number(new Date()) < endTime) {\n\t\t\t\t\t\t\t\tsetTimeout(poll, pollInterval);\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\telse {\n\t\t\t\t\t\t\t\tdone(null);\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t})();\n\t\t\t\t\t}).apply(this, arguments);","args":["return (function (selector, options) {\n      var $matchingEl = $(selector);\n\n      if ($matchingEl.length === 0) {\n        return null;\n      }\n\n      if ($matchingEl.length > 1) {\n        throw new Error('Multiple elements matched. Make a more precise selector - ' + selector);\n      }\n\n      if (! $matchingEl.is(':visible')) {\n        return null;\n      }\n\n      if ($matchingEl.is(':animated')) {\n        // If the element is animating, try again after a delay. Clicks\n        // do not always register if the element is in the midst of\n        // an animation.\n        return null;\n      }\n\n      return true;\n    }).apply(this, arguments);",["#proceed",{}],10000,67]}] Timed out
Build info: version: '3.0.1', revision: '1969d75', time: '2016-10-18 09:48:19 -0700'
System info: host: 'cpe685b35cfb328-cm78cd8e5bb3f5.cpe.net.cable.rogers.com', ip: '99.248.72.46', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.2', java.version: '1.8.0_111'
Driver info: org.openqa.selenium.firefox.FirefoxDriver
Capabilities [{rotatable=false, raisesAccessibilityExceptions=false, marionette=true, firefoxOptions={args=[], profile=UEsDBBQACAgIABqRMEoAAAAAAAAAAAAAAAAHAAAAdXNlci5qc51YXW/bNhR9368w8jC0wEzLSYttKYoha9OhQLsM9YLujaCpK4s1TXIkZdn79bukJMdfouIiebBoUvfjnHvupSsHlhoLxYurudU1PhHHCoifhVqQFZM1s0BAsbmE/OqnUcGkg5dvfqieTubM4x6jrQ9HSmDSl80zwU1rwQfOixyUF35Lig3jXFfKO8IqX5LKCjxxVXpvbieT6fXPJMO/6e2vWZZN1tOr4ddYWGkPxImFEqr3fTfZTTZp9vzWevzWbRX/kWvlYePfFhuag1t6beglZgttOdBkKNH0077Lzed6hShtKC+tXgF13Arjqa0U9WIFaPMmOzzQ4ZzrWknNcjyt2CIAX+r6awlqhmgGJM+DZdCMwRA9m9OVzpkc4AasvdbSEbAWo9TKablPB2+ry9lUCOv8l0qdN7kLUDgj2Rby+2k2+1N7PIkHXh1udiHYypASk2cwDbQGyUMiKysDYEfZRjRAOYFhkMqgp0AUvrjYPuKW89444Ig90gPrSNG1gBrDoiuxgTxmnGoV/bqsrioToLtPZX5nuBAS8BNx3gruqbZiIRQ1Wgq+fY7hZicJi7NqvhIuxP9XXLzjHIzv82CnKG2OQ35x5xEfdVFIoWDMjMGylxJTMt8i3wtWSX+WISvIBSOKrcWCeW2JAdt6FTDvZ9YhFFi1YAMWNbDl+QDQpQ5mh0lU/vy2U1IkS0KBr7VdNrGOYYPuY6WibDyHQDuv4zIMUaiD4P7x0x25IewJr9P0HMPVlUQoAzbXlb+dS6aWRxWxAAUWVSAs4WbliV5jraMqhnOf9X9CSjZ5TbLRi8+MC+W1K9+MPmIYcoQLo4fZ6J/RFKUwezOy69tXKIsvR38AX+rJdTbN8H86+iDQlN5MwpejD5u7v8FhFiZTkp2U5y6ZQdZEgRs76e2vZBRe/b4hzoxrAwGIaY9qouy5iMSDejDQo0B7755LzZdSoBtJThyCLAG5vcP4GcjupfwU145vjc6Pu3pDmH3Vw7nu5e3eHpVoO89wyxnozMi1wc5cmctbYw3z3ApMDWXOVUHTFWYHqZNTlIsqKvZpur6xNWtCItr4CGIoso/qXdO8krUT6dGGFmgVayktAyHmkMcxhqSAR4tjpPE4xBt9POZiUhH6sWq1kSLDFc2reCZ0kUEO8xLL8R22febFXMhg+RUCNCD6JUjZHH3faPnvzTcX9zqpF4uw2A4Pyaa353V3LDVrbIxQCBE6GjiG9WAhwPdvhYLTYyFIKPaaNg9NtLXwJUf3z7erA6BNKVy5HYevDbJyjF1l4Us8d/369flEWmB5mGbAIzkg/xBmnztME7+AinHAHDeZ6BOgWFzIdayukDuCzQ9UsBzmh7DyoGZVDH7YbCOQ76R2Peb2hZoh59dwv1sZPuEqE5jRK6aIJc4CGDus5mBn8XFA6RzEAcLhPAF4MspFgaMu5Za5Mn0WJXJJYlUF+lnFwuR43atFsQfvaREH69M5xaX8QX0VCmd2h70Su6jVj18+pTtzqPmmhCgynH5z6F4wLZpkpKqhLgX259C10sUQRvul8KSleHeTaJUmGZPEOcliz0j3xW63ghqJRdqhPBHz8OSa1I9TSNFyHRN/iumARP7SJ5E7/ebOkFRRdulFLJDL3m5Jc/uK3l8P7R0YOPbrPay1Hy8nwX5lJm3uddaUKJ9GYuFb1L4koRqKhIH1+0etoXH6qYobExR7Bj5SWLfz+9Cl4+D+N6RIzPKyvVEM3HSx0huSkrxamWR29+40z0tTd308vjfe9L43DD7P/6XHgQ/F6gZmwHbXVSq7B1e6ISwPK8DrJahZnLgev3zc8+PYYOL3AsLyXAR5ZQMqdfantiQaaalRYlF62XOZP0yRCzroKdb4M+4WJ8F29wyywkYZ7wVXobnie3psRp7TOKzjzTBehmjT77vFtAdn52ky/Z5D10PlJot2riB7v/38D1BLBwgCT50AtQUAACEVAABQSwECFAAUAAgICAAakTBKAk+dALUFAAAhFQAABwAAAAAAAAAAAAAAAAAAAAAAdXNlci5qc1BLBQYAAAAAAQABADUAAADqBQAAAAA=}, appBuildId=20161208153507, version=50.1.0, platform=MAC, proxy={}, command_id=1, specificationLevel=0, acceptSslCerts=false, processId=7626, browserVersion=50.1.0, platformVersion=16.3.0, name=tests/intern_functional_oauth, XULappId={ec8030f7-c20a-464f-9b0e-13a3a9e97384}, browserName=firefox, takesScreenshot=true, idle-timeout=60, takesElementScreenshot=true, platformName=darwin, firefox_profile=UEsDBBQACAAIABm5MEoAAAAAAAAAA...}]
Session ID: c95a1916-af81-dc45-86c6-dbe6d20d0f5c
  at runRequest  <node_modules/intern/node_modules/leadfoot/Session.js:92:40>
  at <node_modules/intern/node_modules/leadfoot/Session.js:113:39>
  at new Promise  <node_modules/intern/node_modules/dojo/Promise.ts:411:3>
  at ProxiedSession._post  <node_modules/intern/node_modules/leadfoot/Session.js:67:10>
  at ProxiedSession.Session.executeAsync  <node_modules/intern/node_modules/leadfoot/Session.js:736:15>
  at <node_modules/intern/node_modules/leadfoot/helpers/pollUntil.js:90:21>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/node_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:1553:10>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at process._tickCallback  <node.js:365:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Test.registerSuite.signup, verify different browser, verification_redirect=always [as test]  <tests/functional/oauth_sign_up_verification_redirect.js:136:10>
  at <node_modules/intern/lib/Test.js:191:24>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/browser_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/browser_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/browser_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at process._tickCallback  <node.js:365:13>
✓ firefox on any platform - oauth reset password - reset password, verify same browser (4.083s)
✓ firefox on any platform - oauth reset password - reset password, verify same browser with original tab closed (4.392s)
✓ firefox on any platform - oauth reset password - reset password, verify same browser by replacing the original tab (3.766s)
✓ firefox on any platform - oauth reset password - reset password, verify in a different browser, from the original tab's P.O.V. (6.924s)
✓ firefox on any platform - oauth reset password - reset password, verify in a different browser, from the new browser's P.O.V. (4.158s)
✓ firefox on any platform - oauth settings clients - rp listed in apps, can be deleted (8.243s)
✓ firefox on any platform - oauth force_auth - with a registered email (2.188s)
✓ firefox on any platform - oauth force_auth - with an unregistered email (8.289s)
✓ firefox on any platform - oauth force_auth - verified, blocked (2.374s)
✓ firefox on any platform - oauth permissions for untrusted reliers - signin verified (4.72s)
✓ firefox on any platform - oauth permissions for untrusted reliers - re-signin verified, no additional permissions (5.506s)
✓ firefox on any platform - oauth permissions for untrusted reliers - signin unverified, acts like signup (5.268s)
✓ firefox on any platform - oauth permissions for untrusted reliers - signup, verify same browser (10.135s)
✓ firefox on any platform - oauth permissions for untrusted reliers - signup, then signin with no additional permissions (12.478s)
✓ firefox on any platform - oauth permissions for untrusted reliers - signin from signup page (3.441s)
✓ firefox on any platform - oauth permissions for untrusted reliers - signin with new permission available b/c of new account information (8.071s)
× firefox on any platform - oauth permissions for untrusted reliers - signin with additional requested permission (5.736s)
Error: input[name="profile:display_name"] should not be present
  at Command.<anonymous>  <tests/functional/lib/helpers.js:259:19>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at process._tickCallback  <node.js:365:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:258:12>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at process._tickCallback  <node.js:365:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Test.registerSuite.signin with additional requested permission [as test]  <tests/functional/oauth_permissions.js:265:10>
  at <node_modules/intern/lib/Test.js:191:24>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/browser_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/browser_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/browser_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at process._tickCallback  <node.js:365:13>
✓ firefox on any platform - oauth permissions for untrusted reliers - signin after de-selecting a requested permission (8.182s)
✓ firefox on any platform - oauth permissions for trusted reliers - signup without `prompt=consent` (2.682s)
× firefox on any platform - oauth permissions for trusted reliers - signup with `prompt=consent` (29.899s)
NoSuchElement: [POST http://localhost:4444/wd/hub/session/e1ef1ce4-be72-4892-9ca5-79df08429591/element / {"using":"css selector","value":"#fxa-permissions-header"}] Unable to locate element: #fxa-permissions-header
For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html
Build info: version: '3.0.1', revision: '1969d75', time: '2016-10-18 09:48:19 -0700'
System info: host: 'cpe685b35cfb328-cm78cd8e5bb3f5.cpe.net.cable.rogers.com', ip: '99.248.72.46', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.2', java.version: '1.8.0_111'
Driver info: org.openqa.selenium.firefox.FirefoxDriver
Capabilities [{rotatable=false, raisesAccessibilityExceptions=false, marionette=true, firefoxOptions={args=[], profile=UEsDBBQACAgIABqRMEoAAAAAAAAAAAAAAAAHAAAAdXNlci5qc51YXW/bNhR9368w8jC0wEzLSYttKYoha9OhQLsM9YLujaCpK4s1TXIkZdn79bukJMdfouIiebBoUvfjnHvupSsHlhoLxYurudU1PhHHCoifhVqQFZM1s0BAsbmE/OqnUcGkg5dvfqieTubM4x6jrQ9HSmDSl80zwU1rwQfOixyUF35Lig3jXFfKO8IqX5LKCjxxVXpvbieT6fXPJMO/6e2vWZZN1tOr4ddYWGkPxImFEqr3fTfZTTZp9vzWevzWbRX/kWvlYePfFhuag1t6beglZgttOdBkKNH0077Lzed6hShtKC+tXgF13Arjqa0U9WIFaPMmOzzQ4ZzrWknNcjyt2CIAX+r6awlqhmgGJM+DZdCMwRA9m9OVzpkc4AasvdbSEbAWo9TKablPB2+ry9lUCOv8l0qdN7kLUDgj2Rby+2k2+1N7PIkHXh1udiHYypASk2cwDbQGyUMiKysDYEfZRjRAOYFhkMqgp0AUvrjYPuKW89444Ig90gPrSNG1gBrDoiuxgTxmnGoV/bqsrioToLtPZX5nuBAS8BNx3gruqbZiIRQ1Wgq+fY7hZicJi7NqvhIuxP9XXLzjHIzv82CnKG2OQ35x5xEfdVFIoWDMjMGylxJTMt8i3wtWSX+WISvIBSOKrcWCeW2JAdt6FTDvZ9YhFFi1YAMWNbDl+QDQpQ5mh0lU/vy2U1IkS0KBr7VdNrGOYYPuY6WibDyHQDuv4zIMUaiD4P7x0x25IewJr9P0HMPVlUQoAzbXlb+dS6aWRxWxAAUWVSAs4WbliV5jraMqhnOf9X9CSjZ5TbLRi8+MC+W1K9+MPmIYcoQLo4fZ6J/RFKUwezOy69tXKIsvR38AX+rJdTbN8H86+iDQlN5MwpejD5u7v8FhFiZTkp2U5y6ZQdZEgRs76e2vZBRe/b4hzoxrAwGIaY9qouy5iMSDejDQo0B7755LzZdSoBtJThyCLAG5vcP4GcjupfwU145vjc6Pu3pDmH3Vw7nu5e3eHpVoO89wyxnozMi1wc5cmctbYw3z3ApMDWXOVUHTFWYHqZNTlIsqKvZpur6xNWtCItr4CGIoso/qXdO8krUT6dGGFmgVayktAyHmkMcxhqSAR4tjpPE4xBt9POZiUhH6sWq1kSLDFc2reCZ0kUEO8xLL8R22febFXMhg+RUCNCD6JUjZHH3faPnvzTcX9zqpF4uw2A4Pyaa353V3LDVrbIxQCBE6GjiG9WAhwPdvhYLTYyFIKPaaNg9NtLXwJUf3z7erA6BNKVy5HYevDbJyjF1l4Us8d/369flEWmB5mGbAIzkg/xBmnztME7+AinHAHDeZ6BOgWFzIdayukDuCzQ9UsBzmh7DyoGZVDH7YbCOQ76R2Peb2hZoh59dwv1sZPuEqE5jRK6aIJc4CGDus5mBn8XFA6RzEAcLhPAF4MspFgaMu5Za5Mn0WJXJJYlUF+lnFwuR43atFsQfvaREH69M5xaX8QX0VCmd2h70Su6jVj18+pTtzqPmmhCgynH5z6F4wLZpkpKqhLgX259C10sUQRvul8KSleHeTaJUmGZPEOcliz0j3xW63ghqJRdqhPBHz8OSa1I9TSNFyHRN/iumARP7SJ5E7/ebOkFRRdulFLJDL3m5Jc/uK3l8P7R0YOPbrPay1Hy8nwX5lJm3uddaUKJ9GYuFb1L4koRqKhIH1+0etoXH6qYobExR7Bj5SWLfz+9Cl4+D+N6RIzPKyvVEM3HSx0huSkrxamWR29+40z0tTd308vjfe9L43DD7P/6XHgQ/F6gZmwHbXVSq7B1e6ISwPK8DrJahZnLgev3zc8+PYYOL3AsLyXAR5ZQMqdfantiQaaalRYlF62XOZP0yRCzroKdb4M+4WJ8F29wyywkYZ7wVXobnie3psRp7TOKzjzTBehmjT77vFtAdn52ky/Z5D10PlJot2riB7v/38D1BLBwgCT50AtQUAACEVAABQSwECFAAUAAgICAAakTBKAk+dALUFAAAhFQAABwAAAAAAAAAAAAAAAAAAAAAAdXNlci5qc1BLBQYAAAAAAQABADUAAADqBQAAAAA=}, appBuildId=20161208153507, version=50.1.0, platform=MAC, proxy={}, command_id=1, specificationLevel=0, acceptSslCerts=false, processId=7626, browserVersion=50.1.0, platformVersion=16.3.0, name=tests/intern_functional_oauth, XULappId={ec8030f7-c20a-464f-9b0e-13a3a9e97384}, browserName=firefox, takesScreenshot=true, idle-timeout=60, takesElementScreenshot=true, platformName=darwin, firefox_profile=UEsDBBQACAAIABm5MEoAAAAAAAAAA...}]
Session ID: c95a1916-af81-dc45-86c6-dbe6d20d0f5c
*** Element info: {Using=css selector, value=#fxa-permissions-header}
  at runRequest  <node_modules/intern/node_modules/leadfoot/Session.js:92:40>
  at <node_modules/intern/node_modules/leadfoot/Session.js:113:39>
  at new Promise  <node_modules/intern/node_modules/dojo/Promise.ts:411:3>
  at ProxiedSession._post  <node_modules/intern/node_modules/leadfoot/Session.js:67:10>
  at ProxiedSession.Session.find  <node_modules/intern/node_modules/leadfoot/Session.js:1302:15>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:42:36>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at Command.find  <node_modules/intern/node_modules/leadfoot/Command.js:23:10>
  at Command.prototype.(anonymous function) [as findByCssSelector]  <node_modules/intern/node_modules/leadfoot/lib/strategies.js:29:16>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:1273:10>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at process._tickCallback  <node.js:365:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Test.registerSuite.signup with `prompt=consent` [as test]  <tests/functional/oauth_permissions.js:350:10>
  at <node_modules/intern/lib/Test.js:191:24>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/browser_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/browser_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/browser_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at process._tickCallback  <node.js:365:13>
✓ firefox on any platform - oauth permissions for trusted reliers - signin without `prompt=consent` (2.545s)
× firefox on any platform - oauth permissions for trusted reliers - signin with `prompt=consent` (29.748s)
NoSuchElement: [POST http://localhost:4444/wd/hub/session/e1ef1ce4-be72-4892-9ca5-79df08429591/element / {"using":"css selector","value":"#fxa-permissions-header"}] Unable to locate element: #fxa-permissions-header
For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html
Build info: version: '3.0.1', revision: '1969d75', time: '2016-10-18 09:48:19 -0700'
System info: host: 'cpe685b35cfb328-cm78cd8e5bb3f5.cpe.net.cable.rogers.com', ip: '99.248.72.46', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.2', java.version: '1.8.0_111'
Driver info: org.openqa.selenium.firefox.FirefoxDriver
Capabilities [{rotatable=false, raisesAccessibilityExceptions=false, marionette=true, firefoxOptions={args=[], profile=UEsDBBQACAgIABqRMEoAAAAAAAAAAAAAAAAHAAAAdXNlci5qc51YXW/bNhR9368w8jC0wEzLSYttKYoha9OhQLsM9YLujaCpK4s1TXIkZdn79bukJMdfouIiebBoUvfjnHvupSsHlhoLxYurudU1PhHHCoifhVqQFZM1s0BAsbmE/OqnUcGkg5dvfqieTubM4x6jrQ9HSmDSl80zwU1rwQfOixyUF35Lig3jXFfKO8IqX5LKCjxxVXpvbieT6fXPJMO/6e2vWZZN1tOr4ddYWGkPxImFEqr3fTfZTTZp9vzWevzWbRX/kWvlYePfFhuag1t6beglZgttOdBkKNH0077Lzed6hShtKC+tXgF13Arjqa0U9WIFaPMmOzzQ4ZzrWknNcjyt2CIAX+r6awlqhmgGJM+DZdCMwRA9m9OVzpkc4AasvdbSEbAWo9TKablPB2+ry9lUCOv8l0qdN7kLUDgj2Rby+2k2+1N7PIkHXh1udiHYypASk2cwDbQGyUMiKysDYEfZRjRAOYFhkMqgp0AUvrjYPuKW89444Ig90gPrSNG1gBrDoiuxgTxmnGoV/bqsrioToLtPZX5nuBAS8BNx3gruqbZiIRQ1Wgq+fY7hZicJi7NqvhIuxP9XXLzjHIzv82CnKG2OQ35x5xEfdVFIoWDMjMGylxJTMt8i3wtWSX+WISvIBSOKrcWCeW2JAdt6FTDvZ9YhFFi1YAMWNbDl+QDQpQ5mh0lU/vy2U1IkS0KBr7VdNrGOYYPuY6WibDyHQDuv4zIMUaiD4P7x0x25IewJr9P0HMPVlUQoAzbXlb+dS6aWRxWxAAUWVSAs4WbliV5jraMqhnOf9X9CSjZ5TbLRi8+MC+W1K9+MPmIYcoQLo4fZ6J/RFKUwezOy69tXKIsvR38AX+rJdTbN8H86+iDQlN5MwpejD5u7v8FhFiZTkp2U5y6ZQdZEgRs76e2vZBRe/b4hzoxrAwGIaY9qouy5iMSDejDQo0B7755LzZdSoBtJThyCLAG5vcP4GcjupfwU145vjc6Pu3pDmH3Vw7nu5e3eHpVoO89wyxnozMi1wc5cmctbYw3z3ApMDWXOVUHTFWYHqZNTlIsqKvZpur6xNWtCItr4CGIoso/qXdO8krUT6dGGFmgVayktAyHmkMcxhqSAR4tjpPE4xBt9POZiUhH6sWq1kSLDFc2reCZ0kUEO8xLL8R22febFXMhg+RUCNCD6JUjZHH3faPnvzTcX9zqpF4uw2A4Pyaa353V3LDVrbIxQCBE6GjiG9WAhwPdvhYLTYyFIKPaaNg9NtLXwJUf3z7erA6BNKVy5HYevDbJyjF1l4Us8d/369flEWmB5mGbAIzkg/xBmnztME7+AinHAHDeZ6BOgWFzIdayukDuCzQ9UsBzmh7DyoGZVDH7YbCOQ76R2Peb2hZoh59dwv1sZPuEqE5jRK6aIJc4CGDus5mBn8XFA6RzEAcLhPAF4MspFgaMu5Za5Mn0WJXJJYlUF+lnFwuR43atFsQfvaREH69M5xaX8QX0VCmd2h70Su6jVj18+pTtzqPmmhCgynH5z6F4wLZpkpKqhLgX259C10sUQRvul8KSleHeTaJUmGZPEOcliz0j3xW63ghqJRdqhPBHz8OSa1I9TSNFyHRN/iumARP7SJ5E7/ebOkFRRdulFLJDL3m5Jc/uK3l8P7R0YOPbrPay1Hy8nwX5lJm3uddaUKJ9GYuFb1L4koRqKhIH1+0etoXH6qYobExR7Bj5SWLfz+9Cl4+D+N6RIzPKyvVEM3HSx0huSkrxamWR29+40z0tTd308vjfe9L43DD7P/6XHgQ/F6gZmwHbXVSq7B1e6ISwPK8DrJahZnLgev3zc8+PYYOL3AsLyXAR5ZQMqdfantiQaaalRYlF62XOZP0yRCzroKdb4M+4WJ8F29wyywkYZ7wVXobnie3psRp7TOKzjzTBehmjT77vFtAdn52ky/Z5D10PlJot2riB7v/38D1BLBwgCT50AtQUAACEVAABQSwECFAAUAAgICAAakTBKAk+dALUFAAAhFQAABwAAAAAAAAAAAAAAAAAAAAAAdXNlci5qc1BLBQYAAAAAAQABADUAAADqBQAAAAA=}, appBuildId=20161208153507, version=50.1.0, platform=MAC, proxy={}, command_id=1, specificationLevel=0, acceptSslCerts=false, processId=7626, browserVersion=50.1.0, platformVersion=16.3.0, name=tests/intern_functional_oauth, XULappId={ec8030f7-c20a-464f-9b0e-13a3a9e97384}, browserName=firefox, takesScreenshot=true, idle-timeout=60, takesElementScreenshot=true, platformName=darwin, firefox_profile=UEsDBBQACAAIABm5MEoAAAAAAAAAA...}]
Session ID: c95a1916-af81-dc45-86c6-dbe6d20d0f5c
*** Element info: {Using=css selector, value=#fxa-permissions-header}
  at runRequest  <node_modules/intern/node_modules/leadfoot/Session.js:92:40>
  at <node_modules/intern/node_modules/leadfoot/Session.js:113:39>
  at new Promise  <node_modules/intern/node_modules/dojo/Promise.ts:411:3>
  at ProxiedSession._post  <node_modules/intern/node_modules/leadfoot/Session.js:67:10>
  at ProxiedSession.Session.find  <node_modules/intern/node_modules/leadfoot/Session.js:1302:15>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:42:36>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at Command.find  <node_modules/intern/node_modules/leadfoot/Command.js:23:10>
  at Command.prototype.(anonymous function) [as findByCssSelector]  <node_modules/intern/node_modules/leadfoot/lib/strategies.js:29:16>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:1273:10>
  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:543:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at process._tickCallback  <node.js:365:13>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Test.registerSuite.signin with `prompt=consent` [as test]  <tests/functional/oauth_permissions.js:373:10>
  at <node_modules/intern/lib/Test.js:191:24>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/browser_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/browser_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/browser_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/browser_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:436:9>
  at process._tickCallback  <node.js:365:13>
× firefox on any platform - oauth permissions for trusted reliers - signin without `prompt=consent`, then re-signin with `prompt=consent` (30.006s)
CancelError: Timeout reached on firefox on any platform - oauth permissions for trusted reliers - signin without `prompt=consent`, then re-signin with `prompt=consent`
  at null._onTimeout  <node_modules/intern/lib/Test.js:141:20>
  at Timer.listOnTimeout  <timers.js:92:15>
  at Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:542:10>
  at Command.finally  <node_modules/intern/node_modules/leadfoot/Command.js:566:15>
  at <node_modules/intern/lib/Test.js:216:37>
  at Promise.cancel  <node_modules/intern/browser_modules/dojo/Promise.ts:343:37>
  at null._onTimeout  <node_modules/intern/lib/Test.js:143:21>
  at Timer.listOnTimeout  <timers.js:92:15>
✓ firefox on any platform - oauth permissions for trusted reliers - force_auth without `prompt=consent` (2.163s)
✓ firefox on any platform - oauth permissions for trusted reliers - force_auth with `prompt=consent` (3.141s)
✓ firefox on any platform - Sign in with OAuth after Sync - sign in to OAuth with Sync creds (12.771s)
No unit test coverage for firefox on any platform
firefox on any platform: 7/89 tests failed

TOTAL: tested 1 platforms, 7/89 tests failed

```

--- a/tests/functional/lib/helpers.js
+++ b/tests/functional/lib/helpers.js
@@ -708,11 +708,15 @@ define([
         return this.parent
           .then(openPage(endpoint, expectedHeader))
           .then(function () {
-            // > 1 because email must always be specified
-            if (Object.keys(queryParams).length > 1) {
-              return this.parent
-                .then(reOpenWithAdditionalQueryParams(queryParams, expectedHeader));
+            var numQueryParams = Object.keys(queryParams).length;
+            if (! numQueryParams || (numQueryParams === 1 && queryParams.email)) {
+              // email will already have been added, if passed in. email is
+              // not passed in for some functional tests to ensure query parameter
+              // validation is working properly.
+              return;
             }
+            return this.parent
+              .then(reOpenWithAdditionalQueryParams(queryParams, expectedHeader));
           });
       }
 

--- a/tests/functional/lib/helpers.js
+++ b/tests/functional/lib/helpers.js
@@ -708,6 +708,7 @@ define([
         return this.parent
           .then(openPage(endpoint, expectedHeader))
           .then(function () {
+            // > 1 because email must always be specified
             if (Object.keys(queryParams).length > 1) {
               return this.parent
                 .then(reOpenWithAdditionalQueryParams(queryParams, expectedHeader));
@@ -724,7 +725,7 @@ define([
         .then(testElementExists(expectedHeader))
 
         .then(function () {
-          if (Object.keys(queryParams).length > 1) {
+          if (Object.keys(queryParams).length) {
             return this.parent
               .then(reOpenWithAdditionalQueryParams(queryParams, expectedHeader));
           }

